File: Core/MessageDatabaseCatalogue.cs
using KolibSoft.MsgHub.Core.Models;
using KolibSoft.Catalogue.Core;
using Microsoft.EntityFrameworkCore;

namespace KolibSoft.MsgHub.Core;

public class MessageDatabaseCatalogue : DatabaseCatalogue<MessageModel, CatalogueFilters>
{

    public MessageDatabaseCatalogue(DbContext dbContext) : base(dbContext) { }

}

File: docs/lib/message_database_catalogue.js
import { MessageModel } from "./message_model.js";
import { DatabaseCatalogue, DbContext } from "./modules/catalogue.js";

class MessageDatabaseCatalogue extends DatabaseCatalogue {

    /**
     * @param {DbContext} dbContext 
     */
    constructor(dbContext) {
        super(json => new MessageModel(json), dbContext, "message");
    }

}

export {
    MessageDatabaseCatalogue
}